internal/collector/instance.go

```diff
@@ -136,18 +151,22 @@ func AddToPod(
 			}},
 		}
 		container.VolumeMounts = append(container.VolumeMounts, logrotateConfigVolumeMount)
-		outPod.Volumes = append(outPod.Volumes, logrotateConfigVolume)
+		template.Spec.Volumes = append(template.Spec.Volumes, logrotateConfigVolume)
 	}
 
-	if feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+	if feature.Enabled(ctx, feature.OpenTelemetryMetrics) && includeMetricsPort {
```

Comment:
Do we only want to apply the label and add the config queries if we're enabling this pod for metrics?

Reply:
Hmm... In my mind the label simply indicates that the Pod has an otel collector sidecar in it; it could feasibly be used by something other than Prometheus.

Not sure what you mean by "config queries"?

Reply:
Oh, yeah, "config queries" doesn't make sense. I meant the custom queries files -- right now, if you have custom queries files on the spec, we will add them to every collector, e.g., pgbouncer and pgbackrest pods get these mounted, even though we don't need them. 

(And actually, in pgbouncer's case, we don't want them, right? In pgbackrest's case, they just don't get used. But if we try to run a query meant for postgres on pgbouncer, that might lead to an error.)

Reply:
Ah, you're talking about the adding of the custom queries configmaps to the config volume as projected sources... Hmm... I don't think that it's a problem that they are added even if the pod doesn't have metrics, but I guess it makes sense to limit where we put them. I'll change that.

Reply:
This also makes me realize that we'll be projecting these custom query configmaps into pods that will be serving metrics but won't be using the custom queries (like pgbouncer), but I'm not sure that it's worth the effort to prevent that.

Reply:
Oh, right, and it won't lead to an error in pgbouncer, because we aren't adding the files to pgbouncer_metrics -- and when we do want to open pgbouncer up to custom metrics, we probably will add a `component` tag to api for each file. OK, I've talked myself back down from being worried.